src/electionguard-ui/ElectionGuard.UI/ViewModels/CreateMultiTallyViewModel.cs

```diff
@@ -0,0 +1,353 @@
+
+    private TallyService _tallyService;
+    private ElectionService _electionService;
+    private ManifestService _manifestService;
+    private KeyCeremonyService _keyCeremonyService;
+    private BallotUploadService _ballotUploadService;
+    private BallotService _ballotService;
+    private TallyJoinedService _tallyJoinedService;
+    private MultiTallyService _multiTallyService;
+
+    public CreateMultiTallyViewModel(
+        IServiceProvider serviceProvider,
+        TallyService tallyService,
+        ManifestService manifestService,
+        ElectionService electionService,
+        KeyCeremonyService keyCeremonyService,
+        BallotUploadService ballotUploadService,
+        BallotService ballotService,
+        TallyJoinedService tallyJoinedService,
+        MultiTallyService multiTallyService) : base("CreateMultiTally", serviceProvider)
+    {
+        _tallyService = tallyService;
+        _manifestService = manifestService;
+        _electionService = electionService;
+        _keyCeremonyService = keyCeremonyService;
+        _ballotUploadService = ballotUploadService;
+        _ballotService = ballotService;
+        _tallyJoinedService = tallyJoinedService;
+        _multiTallyService = multiTallyService;
+        _ = Task.Run(FillKeyCeremonies);
+    }
+
+    [ObservableProperty]
+    private string? _errorMessage;
+
+    [ObservableProperty]
+    private bool _selectAll;
+
+    [ObservableProperty]
+    private string _multiTallyId = string.Empty;
+
+    [ObservableProperty]
+    private bool _electionsLoaded;
+
+    [ObservableProperty]
+    private ObservableCollection<KeyCeremonyRecord> _keyCeremonies = new();
+
+    [ObservableProperty]
+    private KeyCeremonyRecord _selectedKeyCeremony;
+
+    [ObservableProperty]
+    private MultiTallyRecord? _currentMultiTally;
+
+    [ObservableProperty]
+    [NotifyCanExecuteChangedFor(nameof(CreateTalliesCommand))]
+    [NotifyCanExecuteChangedFor(nameof(JoinTalliesCommand))]
+    private string _currentResultsPath = string.Empty;
+
+    [ObservableProperty]
+    private ObservableCollection<ElectionItem> _elections = new();
+
+    [ObservableProperty]
+    private ObservableCollection<object> _selectedElections = new();
+
+    partial void OnMultiTallyIdChanged(string value)
+    {
+        if (string.IsNullOrEmpty(value))
+        {
+            CurrentMultiTally = null;
+        }
+
+        _ = Shell.Current.CurrentPage.Dispatcher.DispatchAsync(async () =>
+        {
+            // load the elections that are in the multitally
+            var multiTally = await _multiTallyService.GetByMultiTallyIdAsync(value);
+            if (multiTally != null)
+            {
+                foreach (var tallyId in multiTally.TallyIds)
+                {
+                    var tally = await _tallyService.GetByTallyIdAsync(tallyId);
+                    var election = await _electionService.GetByElectionIdAsync(tally!.ElectionId!);

```

Comment:
could be a future enhancement if needed for performance.  This should not be an big performance issue since it is only when the guardians load the multi tally selection page.